File: java/src/main/java/io/fivexer/sdk/Notifications.java
package io.fivexer.sdk;

/** Notification sequences (when to notify) and channels (where to deliver). */
public final class Notifications {

    private final NotificationSequences sequencesResource;
    private final NotificationChannels channelsResource;

    Notifications(Fivexer client) {
        this.sequencesResource = new NotificationSequences(client);
        this.channelsResource = new NotificationChannels(client);
    }

    public NotificationSequences sequences() { return sequencesResource; }

    public NotificationChannels channels() { return channelsResource; }
}

File: php/src/Resource/Notifications.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Resource;

use Fivexer\SDK\Fivexer;

/** Notification sequences (when to notify) and channels (where to deliver). */
final class Notifications
{
    private readonly NotificationSequences $sequencesResource;
    private readonly NotificationChannels $channelsResource;

    public function __construct(Fivexer $client)
    {
        $this->sequencesResource = new NotificationSequences($client);
        $this->channelsResource = new NotificationChannels($client);
    }

    public function sequences(): NotificationSequences
    {
        return $this->sequencesResource;
    }

    public function channels(): NotificationChannels
    {
        return $this->channelsResource;
    }
}
